File: app/src/main/java/com/lamnt/foodorder/view/adapter/pageradapter/IntroPagerAdapter.java
package com.lamnt.foodorder.view.adapter.pageradapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.lamnt.foodorder.view.fragment.intro.IntroItemFragment;

public class IntroPagerAdapter extends FragmentStateAdapter {
    private OnTabSelected onTabSelected;

    public IntroPagerAdapter(@NonNull FragmentActivity fragmentActivity,
                             OnTabSelected onTabSelected) {
        super(fragmentActivity);
        this.onTabSelected = onTabSelected;
    }


    @NonNull
    @Override
    public Fragment createFragment(int position) {
        onTabSelected.onItemSelected(position);
        if (position == 0) {
            return IntroItemFragment.newInstance(IntroItemFragment.TYPE_ORDER);
        } else if (position == 1) {
            return IntroItemFragment.newInstance(IntroItemFragment.TYPE_DELIVERY);
        } else {
            return IntroItemFragment.newInstance(IntroItemFragment.TYPE_PAYMENT);
        }
    }

    @Override
    public int getItemCount() {
        return 3;
    }

    public interface OnTabSelected{
        void onItemSelected(int position);
    }
}

File: app/src/main/java/com/lamnt/foodorder/view/adapter/pageradapter/IntroPagerAdapter.kt
package com.lamnt.foodorder.view.adapter.pageradapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.lamnt.foodorder.view.fragment.intro.IntroItemFragment

class IntroPagerAdapter(fragmentActivity: FragmentActivity,
                        private val onTabSelected: OnTabSelected) : FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        onTabSelected.onItemSelected(position)
        return when (position) {
            0 -> {
                IntroItemFragment.newInstance(IntroItemFragment.TYPE_ORDER)
            }
            1 -> {
                IntroItemFragment.newInstance(IntroItemFragment.TYPE_DELIVERY)
            }
            else -> {
                IntroItemFragment.newInstance(IntroItemFragment.TYPE_PAYMENT)
            }
        }
    }

    override fun getItemCount(): Int {
        return 3
    }

    interface OnTabSelected {
        fun onItemSelected(position: Int)
    }

}